contracts/strategies/StrategyCurveGUSDVoterProxy.sol

```diff
@@ -0,0 +1,216 @@
+    }
+
+    function setProxy(address _proxy) external {
+        require(msg.sender == governance, "!governance");
+        proxy = _proxy;
+    }
+
+    function switchDex() external {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
+        // save gas and storage
+        if (dex == address(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D)) {
+            dex = address(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F);
+        } else {
+            dex = address(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D);
+        }
+    }
+
+    function deposit() public {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            IERC20(want).safeTransfer(proxy, _want);
+            IVoterProxy(proxy).deposit(gauge, want);
+        }
+    }
+
+    // Controller only function for creating additional rewards from dust
+    function withdraw(IERC20 _asset) external returns (uint256 balance) {
+        require(msg.sender == controller, "!controller");
+        require(want != address(_asset), "want");
+        require(crv != address(_asset), "crv");
+        require(dai != address(_asset), "dai");
+        balance = _asset.balanceOf(address(this));
+        _asset.safeTransfer(controller, balance);
+    }
+
+    // Withdraw partial funds, normally used with a vault withdrawal
+    function withdraw(uint256 _amount) external {
+        require(msg.sender == controller, "!controller");
+        uint256 _balance = IERC20(want).balanceOf(address(this));
+        if (_balance < _amount) {
+            _amount = _withdrawSome(_amount.sub(_balance));
+            _amount = _amount.add(_balance);
+        }
+
+        uint256 _fee = _amount.mul(withdrawalFee).div(FEE_DENOMINATOR);
+
+        IERC20(want).safeTransfer(IController(controller).rewards(), _fee);
+        address _vault = IController(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        IERC20(want).safeTransfer(_vault, _amount.sub(_fee));
+    }
+
+    function _withdrawSome(uint256 _amount) internal returns (uint256) {
+        return IVoterProxy(proxy).withdraw(gauge, want, _amount);
+    }
+
+    // Withdraw all funds, normally used when migrating strategies
+    function withdrawAll() external returns (uint256 balance) {
+        require(msg.sender == controller, "!controller");
+        _withdrawAll();
+
+        balance = IERC20(want).balanceOf(address(this));
+
+        address _vault = IController(controller).vaults(address(want));
+        require(_vault != address(0), "!vault"); // additional protection so we don't burn the funds
+        IERC20(want).safeTransfer(_vault, balance);
+    }
+
+    function _withdrawAll() internal {
+        IVoterProxy(proxy).withdrawAll(gauge, want);
+    }
+
+    function harvest() public {
+        require(msg.sender == strategist || msg.sender == governance, "!authorized");
```

Comment:
Add keeper role for the Keep3rJob :)

Reply:
i think we discussed this in another PR but for v1, the latest strategies don't consider a keeper role since keepers AFAIK could be outside parties and we don't want to risk flash loan deposit + harvest + withdraw related issues, but i maybe remembering incorrectly.

Reply:
This needs more discussion. I'll draft the pros and cons later.